airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_message.yaml

```diff
@@ -118,10 +118,43 @@ definitions:
       name:
         type: string
         description: Stream's name.
-      json_schema:
-        description: Stream schema using Json Schema specs.
+      discovered:
+        description: Contains information about the data in the source. This field is set by the source and should be read-only thereafter.
         type: object
-        existingJavaType: com.fasterxml.jackson.databind.JsonNode
+        properties:
+          json_schema:
+            description: Stream schema using Json Schema specs.
+            type: object
+            existingJavaType: com.fasterxml.jackson.databind.JsonNode
+          metadata:
+            description: Additional information about the data in the source, including information about what Airbyte features it supports.
+            type: object
+            properties:
+              supported_sync_modes:
+                type: array
+                items:
+                  "$ref": "#/definitions/SyncMode"
+                # backwards compatibility: if not present, assume full_refresh only.
+                default:
+                  [full_refresh]
+              default_comparable:
+                description: Path to the field that will be used to determine if a record is new or modified since the last sync. If not provided by the source, the end user will have to specify the comparable themselves.
+                type: array
+                items: string
+      configuration:
+        description: Contains information about how the data should be synced. this section is configured by the end-user and is never touched by the source.
+        properties:
+          # backwards compatibility: if not present, assume full_refresh.
+          sync_mode:
+            "$ref": "#/definitions/SyncMode"
+            default: full_refresh
+          comparable:
```

Comment:
would prefer to use a name that indicates what it's used for, something like `state_bookmark` or `state_cursor` or even just `state`. `comparable` is an overly specific term to describe this imo. nothing stopping a source from having this be whatever information they want e.g: json

Reply:
went with cursor.